elephas/utils/sockets.py

```diff
@@ -9,8 +9,9 @@ def determine_master(port=4000):
     :param port: port on which the application runs
     :return: Master address
     """
-    return gethostbyname(gethostname()) + ":" + str(port)
-
+    import os
+    return os.environ['SPARK_LOCAL_IP'] + ":" + str(port)
```

Comment:
please move the import to the top of the file.

Also, this will only work for your use case, since you provide your local IP through an environment variable. Can you please create a check like: 

```
if os.environ['SPARK_LOCAL_IP']:
    return os.environ['SPARK_LOCAL_IP'] + ":" + str(port)
else:
    retrun gethostbyname(gethostname()) + ":" + str(port)
```